src/API/PluginRoutes.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace CF\API;
+
+class PluginRoutes
+{
+    public static $routes = array(
+        'account' => array(
+            'class' => 'CF\API\AbstractPluginActions',
+            'methods' => array(
+                'POST' => array(
+                    'function' => 'login',
+                ),
+            ),
+        ),
+
+        'plugin/:id/settings' => array(
+            'class' => 'CF\API\AbstractPluginActions',
+            'methods' => array(
+                'GET' => array(
+                    'function' => 'getPluginSettings',
+                ),
+            ),
+        ),
+
+        'plugin/:id/settings/:human_readable_id' => array(
+            'class' => 'CF\API\AbstractPluginActions',
+            'methods' => array(
+                'PATCH' => array(
+                    'function' => 'patchPluginSettingsRouter',
+                ),
+            ),
+        ),
+    );
+}
```

Comment:
Ideally they won't need to add new routes but yea any extra logic that needs to happen around routes would go here.  

I don't particularly like this solution either so if you have another suggestion I'm all ears.